### PR TITLE
ra-DSPDC-1285-history-check

### PR DESCRIPTION
**Pull request by raaidbroad**
check to see if the processing history table has an entry from _today_; if it does, then we need to diff it and notify clingen. If it doesn't, we good. No, there will not be any instances where this lasts so long that the check should return true but the date is now the next date.

### CODE REVIEW

File: orchestration/templates/export-diff.yaml [hunk 1]
```diff
@@ -25,11 +25,7 @@ spec:
         tasks:
           {{- if $versionIsPinned }}
           - name: check-processing-history
```
Comment (pshapiro4broad) [inline orchestration/templates/export-diff.yaml:27]: maybe rename?
```suggestion
          - name: check-processing-needed
```

**Review by quazi-broad — APPROVED** [review]
Logic looks good to me, approving!
A few questions - they might be dumb - does timing matter for this? I would think that we'd want to check for files at the end of the day depending on when they usually arrive. I'm also wondering if it's possible for this to fail in such a way where we missed sending a notification but by the time we are checking again, it's already a new day?

**Comment by raaidbroad** [comment]
> Logic looks good to me, approving!
> A few questions - they might be dumb - does timing matter for this? I would think that we'd want to check for files at the end of the day depending on when they usually arrive. I'm also wondering if it's possible for this to fail in such a way where we missed sending a notification but by the time we are checking again, it's already a new day?

@quazi-broad We run this workflow at 1pm everyday, and at worst it takes like 2 hours so we're not at risk of crossing days. And nope, the second case you describe won't happen. Timing isn't really an issue here.

**Review by pshapiro4broad — APPROVED** [review]
Looks OK, some minor bash suggestions. If you aren't using it already, I recommend `shellcheck` to check for bash code issues.

**Pull request merged**